pallets/aleph/src/lib.rs

```diff
@@ -121,6 +161,49 @@ pub mod pallet {
         pub(crate) fn set_next_emergency_finalizer(emergency_finalizer: T::AuthorityId) {
             <NextEmergencyFinalizer<T>>::put(emergency_finalizer);
         }
+
+        pub(crate) fn current_session() -> u32 {
+            T::SessionInfoProvider::current_session()
+        }
+
+        // If a scheduled future version change is rescheduled to a different session,
+        // it is possible to reschedule it with the same version as initially.
+        // To cancel a future version change, reschedule it with the current version.
+        // If a scheduled version change has moved into the past, `SessionManager` records it
+        // as the current version.
+        pub(crate) fn schedule_next_aleph_bft_version_change(
+            version_change: VersionChange,
+        ) -> Result<(), &'static str> {
+            let current_session = Self::current_session();
+
+            let session_to_schedule = version_change.session;
+
+            if session_to_schedule < current_session {
+                return Err("Cannot schedule AlephBFT version changes for sessions in the past!");
+            } else if session_to_schedule < current_session + 2 {
+                return Err(
+                    "Tried to schedule an AlephBFT version change less than 2 sessions in advance!",
+                );
+            }
+
+            // Update the scheduled version change with the supplied version change.
+            <AlephBFTScheduledVersionChange<T>>::put(version_change);
+
+            Ok(())
+        }
+
+        pub fn next_session_aleph_bft_version() -> Version {
+            let next_session = Self::current_session() + 1;
+            let scheduled_version_change = Self::aleph_bft_version_change();
+
+            if let Some(version_change) = scheduled_version_change {
+                if next_session == version_change.session {
```

Comment:
An edge case comes to mind that on session boundary the session number changed and somehow the version change was not applied yet.

I'm 95% certain this is now correct, but just wanted to make sure you are also aware and certain :) 

Feel free to resolve after you familiarize yourself with the comment.